runtime/lib/binary.cpp

```diff
@@ -102,6 +104,74 @@ std::vector<TensorDesc> getProgramOutputs(Flatbuffer binary,
 
 } // namespace ttnn
 
+namespace metal {
```

Comment:
They look the same, but the flatbuffer generated functions they call are different and flatbuffer controls the binary layout so they really cannot be mixed.  This can be a subtle gotcha.